PukingPredator/Assets/Scripts/Item.cs

```diff
@@ -1,74 +1,117 @@
 using UnityEngine;
 
+public enum ItemState
+{
+    beingConsumed,
+    inInventory,
+    beingPuked,
+}
 public class Item : MonoBehaviour
 {
-    private GameObject prefab;
-    private Rigidbody rigidBody;
+    /// <summary>
+    /// The lerp factor used when shrinking items.
+    /// </summary>
+    [SerializeField]
+    private float consumptionRate = 0.025f;
 
-    private Vector3 position;
-    private Quaternion rotation;
+    /// <summary>
+    /// The relative scale at which an object will be treated as consumed.
+    /// ie 0.05 means 5% of original size.
+    /// </summary>
+    private float consumptionCutoff = 0.05f;
 
-    private Vector3 playerPos;
+    /// <summary>
+    /// The initial scale of the instance before being eaten. Used to restore
+    /// its size to normal after being puked.
+    /// </summary>
     private Vector3 initialScale;
-    
-    public bool isCollecting = false;
-    float collectionTimer = 0f;
 
-    public void PlaceItem()
-    {
-        if (isCollecting) {  return; }
-        prefab.SetActive(true);
-    }
+    /// <summary>
+    /// The game object for this item in the inventory. It may be inactive or
+    /// active.
+    /// </summary>
+    private GameObject instance;
 
-    public void MoveItem(Vector3 position)
-    {
-        prefab.transform.position = position;
-    }
+    /// <summary>
+    /// The entity that is holding the item.
+    /// </summary>
+    private GameObject owner;
 
-    public void Initialize(GameObject newPrefab, Vector3 newPosition, Quaternion newRotation)
-    {
-        prefab = newPrefab;
-        initialScale = prefab.transform.localScale;
-        rigidBody = prefab.GetComponent<Rigidbody>();
+    /// <summary>
+    /// If the item is entering, leaving, or sitting in the inventory.
+    /// </summary>
+    public ItemState state { get; private set; } = ItemState.beingConsumed;
 
-        position = newPosition;
-        rotation = newRotation == default ? Quaternion.identity : newRotation;
-    }
 
-    public void Collect(Vector3 pos)
-    {
-        playerPos = pos;
-        isCollecting = true;
-    }
 
     public void Update()
     {
-        if (isCollecting)
+        switch(state)
         {
-            collectionTimer += Time.deltaTime;
-            float lerpFactor = collectionTimer / 0.6f;
-
-            prefab.transform.position = Vector3.Lerp(prefab.transform.position, playerPos, lerpFactor);
-            prefab.transform.localScale = Vector3.Lerp(prefab.transform.localScale, Vector3.zero, lerpFactor);
-
-            if (lerpFactor >= 0.8f)
-            {
-                collectionTimer = 0f;
-                prefab.SetActive(false);
+            case ItemState.inInventory:
+                return;
 
-                prefab.transform.localScale = initialScale;
-                prefab.transform.rotation = Quaternion.identity;
+            case ItemState.beingConsumed:
+                var ownerPosition = owner.transform.position;
```

Comment:
For clarity you can have the type instead of just var, theres multiple instances of just var in the files

Reply:
i feel that the type is fairly obvious in each of the cases where i used var. its helpful when dealing with more complex cases like explicitly saying "this is List<Dictionary<String, String>>", but in these cases it doesnt add much and you can always hover over the variable to see the type anyways if youre using a decent IDE. if we think this is an issue then i can be explicit with each variable type